src/ode_solver/explicit_ode_solver.cpp

```diff
@@ -14,7 +14,22 @@ RungeKuttaODESolver<dim,real,n_rk_stages, MeshType>::RungeKuttaODESolver(std::sh
 
 template <int dim, typename real, int n_rk_stages, typename MeshType> 
 void RungeKuttaODESolver<dim,real,n_rk_stages,MeshType>::step_in_time (real dt, const bool pseudotime)
-{  
+{
+    using limiter_enum = Parameters::LimiterParam::LimiterType;
+    limiter_enum limiter_type = this->all_parameters->limiter_param.bound_preserving_limiter;
+    bool apply_tvb = this->all_parameters->limiter_param.use_tvb_limiter;
+    // Apply limiter at every RK stage
+    if ( limiter_type != limiter_enum::none || apply_tvb == true) {
```

Comment:
Previously, when implementing the maximum principle limiter I ran into issues where it would only work for RK4 and not RK3 and the only way to fix it was with the extra application of the limiter at the beginning (ie. limiting the previous solution). In later development however, this wasn't an issue and I forgot to remove that extra application at the beginning. It has been removed in my latest changes which I will push soon. I reran my OOA tests to ensure that the limiter still attains the correct order of accuracy.

After assembling the solution, the limiter does need to be applied otherwise the principles will not be preserved. I ran a few tests without applying after assembling the solution and the cases failed.

Reply:
Sounds good! Thanks for doing this additional testing.